core/store/src/db/refcount.rs

```diff
@@ -161,15 +128,14 @@ where
 }
 
 impl RocksDB {
-    /// DBCol::State has refcounted values.
     /// Merge adds refcounts, zero refcount becomes empty value.
     /// Empty values get filtered by get methods, and removed by compaction.
     pub(crate) fn refcount_merge(
         _new_key: &[u8],
         existing: Option<&[u8]>,
         operands: &rocksdb::MergeOperands,
     ) -> Option<Vec<u8>> {
-        self::refcount_merge(existing, operands)
+        Some(self::refcount_merge(existing, operands))
```

Comment:
Makes me wonder: should we return `None` in some cases? Would that remove the value as I'd expect? It seems like that could be potentially important, to gc keys?

Reply:
Also perhaps worth filing separately, as that would change the behavior in, *hopefully*, non-observable way.

Reply:
I did look into that but got failures.  I’m not sure if returning None from merge operator does what we think it does. ;)  I want to take closer look at it separately.